xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-default/src/main/java/org/xwiki/notifications/internal/WikiNotificationDisplayer.java

```diff
@@ -0,0 +1,198 @@
+     */
+    public WikiNotificationDisplayer(DocumentReference authorReference, TemplateManager templateManager,
+            ScriptContextManager scriptContextManager, ComponentManager componentManager, BaseObject baseObject)
+            throws NotificationException
+    {
+        this.objectReference = baseObject.getReference();
+        this.authorReference = authorReference;
+        this.templateManager = templateManager;
+        this.scriptContextManager = scriptContextManager;
+        this.componentManager = componentManager;
+
+        this.eventType = this.extractProperty(baseObject, WikiNotificationDisplayerDocumentInitializer.EVENT_TYPE);
+        this.supportedEvents = Arrays.asList(this.eventType);
+
+        // Create the template from the given BaseObject property
+        try {
+            this.notificationTemplate = templateManager.createStringTemplate(
+                    this.extractProperty(baseObject,
+                            WikiNotificationDisplayerDocumentInitializer.NOTIFICATION_TEMPLATE),
+                    this.getAuthorReference());
+        } catch (Exception e) {
+            throw new NotificationException(
+                    String.format("Unable to render the template provided in the base object [%s]",
+                            baseObject), e);
+        }
+    }
+
+    /**
+     * Extract the the given property value from the given XObject.
+     *
+     * @param baseObject the XObject that should contain the parameters
+     * @param propertyName the value of the property that should be extracted
+     * @throws NotificationException if an error occurred while extracting the parameter from the base object
+     */
+    private String extractProperty(BaseObject baseObject, String propertyName) throws NotificationException
+    {
+        try {
+            return baseObject.getStringValue(propertyName);
+        } catch (Exception e) {
+            throw new NotificationException(
+                    String.format("Unable to extract the parameter [%s] from the [%s] NotificationDisplayerClass.",
+                            propertyName, baseObject), e);
+        }
+    }
+
+    @Override
+    public Block renderNotification(CompositeEvent eventNotification) throws NotificationException
+    {
+        // Save the old value in the context that refers to EVENT_BINDING_NAME
+        Object oldContextAttribute = scriptContextManager.getCurrentScriptContext().getAttribute(EVENT_BINDING_NAME,
+                ScriptContext.ENGINE_SCOPE);
+
+        try {
+            // Allow the template to access the event during its execution
+            scriptContextManager.getCurrentScriptContext().setAttribute(EVENT_BINDING_NAME, eventNotification,
+                    ScriptContext.ENGINE_SCOPE);
+
+            // If we have no template defined, fallback on the default displayer
+            if (StringUtils.isBlank(this.notificationTemplate.getContent().getContent())) {
```

Comment:
Instead, I would create notificationTemplate only if the content is not empty and test for null notificationTemplate here.